app/src/main/java/com/eis/phoneringer/RingCommands/RingtoneHandler.java

```diff
@@ -0,0 +1,111 @@
+package com.eis.phoneringer.RingCommands;
+
+import android.content.Context;
+import android.media.AudioAttributes;
+import android.media.Ringtone;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+/**
+ * Class that interfaces with the RingtoneManager class of the Android library
+ *
+ * @author Alessandra Tonin, Luca Crema, Alberto Ursino
+ */
+public class RingtoneHandler {
+
+    /**
+     * defaultRingtone instantiated in getDefaultTone
+     */
+    private static Ringtone defaultRingtone = null;
+
+    /**
+     * Instance of the class that is instantiated in getInstance method
+     */
+    private static RingtoneHandler instance = null;
+
+    /**
+     * Private constructor
+     */
+    private RingtoneHandler() {
+    }
+
+    /**
+     * @return the RingtoneHandler instance
+     */
+    public static RingtoneHandler getInstance() {
+        if (instance == null)
+            instance = new RingtoneHandler();
+        return instance;
+    }
+
+    /**
+     * @param context of the application
+     * @param type    of the wanted tone
+     * @return the requested tone
+     */
+    public Ringtone getDefaultTone(Context context, int type) {
+        return RingtoneManager.getRingtone(context, getDefaultToneUri(type));
+    }
+
+    /**
+     * @param context of the application
+     * @return the actual ringtone
```

Comment:
Seems to me that this method doesn't actually return the current system ringtone: if the user calls getDefaultRingtone for the first time variable defaultRingtone is updated. But if in the meantime, the device's Default ringtone is updated and the user calls the method again after that, it will return an outdated Ringtone.

Reply:
You _might_ be right, it all depends on what happens when you try to change the ringtone. Does this class get destroyed? If it does then when the app is re-opened this starts back as it was never instantiated, getting the updated ringtone. Else if this stays the same, then you're right, you would get the old ringtone.

Reply:
You could either change the name (and description) of the method or update defaultRingtone each time the method is called.

Reply:
I honestly don't think there's a problem (at least for now) so I will approve this and add an issue about this when it gets merged